### PR TITLE
CPM-448: Refactor and prepare queries for decoration

### PR DESCRIPTION
**Pull request by amigaire**
**Description (for Contributor and Core Developer)**

Refactor `get/countProduct[Model]WithRemovedAttributes` queries used in the cleaning removed attribute values job, in order to allow decorating theme

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: src/Akeneo/Pim/Enrichment/Bundle/Storage/ElasticsearchAndSql/ProductAndProductModel/CountProductModelsWithRemovedAttribute.php [hunk 1]
```diff
@@ -2,54 +2,53 @@
...
+        ]);
+        foreach ($attributesCodes as $attributeCode){
+            $this->searchQueryBuilder->addShould([
+                'exists' => ['field' => sprintf('values.%s-*', $attributeCode)],
+            ]);
+        }
+
+        $body = $this->searchQueryBuilder->getQuery();
+        unset($body['_source']);
+        unset($body['sort']);
```
Comment (mmetayer) [inline src/Akeneo/Pim/Enrichment/Bundle/Storage/ElasticsearchAndSql/ProductAndProductModel/CountProductModelsWithRemovedAttribute.php:40]: the SearchQueryBuilder automatically adds `_source` and `sort` options, which are incompatible with a `count` ES query.

**Review by mmetayer — APPROVED** [review]

File: src/Akeneo/Pim/Enrichment/Bundle/Storage/ElasticsearchAndSql/ProductAndProductModel/CountProductModelsWithRemovedAttribute.php [hunk 2]
```diff
@@ -2,54 +2,53 @@
...
 
         $result = $this->elasticsearchClient->count($body);
 
+        // Reset query
+        $this->searchQueryBuilder = new SearchQueryBuilder();
```
Comment (cemorin) [inline src/Akeneo/Pim/Enrichment/Bundle/Storage/ElasticsearchAndSql/ProductAndProductModel/CountProductModelsWithRemovedAttribute.php:45]: Remove

File: src/Akeneo/Pim/Enrichment/Bundle/Storage/ElasticsearchAndSql/ProductAndProductModel/CountProductsWithRemovedAttribute.php [hunk 3]
```diff
@@ -2,54 +2,53 @@
...
+        }
+
+        $body = $this->searchQueryBuilder->getQuery();
+        unset($body['_source']);
+        unset($body['sort']);
 
         $result = $this->elasticsearchClient->count($body);
 
+        // Reset query
+        $this->searchQueryBuilder = new SearchQueryBuilder();
```
Comment (cemorin) [inline src/Akeneo/Pim/Enrichment/Bundle/Storage/ElasticsearchAndSql/ProductAndProductModel/CountProductsWithRemovedAttribute.php:45]: Remove

File: src/Akeneo/Pim/Enrichment/Bundle/Storage/ElasticsearchAndSql/ProductAndProductModel/GetProductIdentifiersWithRemovedAttribute.php [hunk 4]
```diff
@@ -60,5 +51,13 @@ public function nextBatch(array $attributesCodes, int $batchSize): iterable
             $body['search_after'] = end($rows['hits']['hits'])['sort'];
             $rows = $this->elasticsearchClient->search($body);
         }
+
+        // Reset query
+        $this->searchQueryBuilder = new SearchQueryBuilder();
```
Comment (cemorin) [inline src/Akeneo/Pim/Enrichment/Bundle/Storage/ElasticsearchAndSql/ProductAndProductModel/GetProductIdentifiersWithRemovedAttribute.php:56]: Remove

File: src/Akeneo/Pim/Enrichment/Bundle/Storage/ElasticsearchAndSql/ProductAndProductModel/GetProductModelIdentifiersWithRemovedAttribute.php [hunk 5]
```diff
@@ -65,5 +51,13 @@ public function nextBatch(array $attributesCodes, int $batchSize): iterable
             $body['search_after'] = end($rows['hits']['hits'])['sort'];
             $rows = $this->elasticsearchClient->search($body);
         }
+
+        // Reset query
+        $this->searchQueryBuilder = new SearchQueryBuilder();
```
Comment (cemorin) [inline src/Akeneo/Pim/Enrichment/Bundle/Storage/ElasticsearchAndSql/ProductAndProductModel/GetProductModelIdentifiersWithRemovedAttribute.php:56]: Remove

**Review by nmarniesse — APPROVED** [review]

**Pull request merged**